doc/cprover-manual/contracts-decreases.md

```diff
@@ -1,3 +1,183 @@
-# Decreases Clause
+# Decreases Clauses
 
-TODO: Document `__CPROVER_decreases`
+A _decreases_ clause specifies a measure that must strictly decrease
+monotonically at every iteration of a loop.
+By demonstrating that the measure
+1. is bounded from below, and
+2. strictly decreases monotonically at each iteration,
+we can prove termination of loops.
+This is because the measure must eventually hit the lower bound
+at which point the loop must terminate,
+since the measure cannot strictly decrease further.
+
+In the scientific literature, this measure is also called a _loop variant_ or _ranking function_.
+Although they can be arbitrary (strict) well-founded relations on arbitrary domains,
+we restrict decreases clauses to (strict) arithmetic inequalities.
+Unlike mathematical integers, arithmetic types in C are all bounded from below.
+Furthermore, unlike mathematical real numbers, floating-point arithmetic types in C are not dense.
+Hence, any strictly decreasing arithmetic expression in C can be used as a decreases clause.
+
```

Comment:
Echoing Daniel's comments to Felipe on earlier documents, I wouldn't try to explain the science here. Maybe we write a background document for folks, but people come to a man-page type document like this for a short and sweet description of what a feature does and how should they use it. An explanation of how/why it works is done better in another format.

Reply:
Thanks; makes sense. This paragraph could be removed.